python/tests/unit/processes/local_runtime/test_local_step.py

```diff
@@ -0,0 +1,558 @@
+    if all parameters are provided."""
+    mock_kernel = AsyncMock(spec=Kernel)
+    mock_kernel.invoke = AsyncMock(return_value=MagicMock(value="result"))
+
+    mock_state = KernelProcessStepState(name="my_step", id="123", state=None)
+    mock_info = KernelProcessStepInfo(inner_step_type=AsyncMock(spec=type), state=mock_state, output_edges={})
+
+    # Create a function that requires one parameter
+    mock_function = AsyncMock(spec=KernelFunction)
+    mock_function.name = "func"
+    mock_function.plugin_name = "test_plugin"
+
+    mock_function.metadata = AsyncMock(spec=KernelFunctionMetadata)
+    mock_function.metadata.name = "func"
+    mock_function.metadata.plugin_name = "test_plugin"
+    mock_function.metadata.is_prompt = False
+
+    step = LocalStep(
+        kernel=mock_kernel,
+        step_info=mock_info,
+        event_namespace="ns",
+        step_state=mock_state,
+        factories={},
+        functions={
+            "func": mock_function,
+        },
+        inputs={"func": {"param": None}},
+        initial_inputs={"func": {"param": None}},
+        initialize_task=True,
+    )
+
+    with patch.object(LocalStep, "emit_event") as mock_emit_event:
+        msg = LocalMessage(
+            source_id="source",
+            destination_id="dest",
+            function_name="func",
+            values={"param": "value"},
+        )
+
+        await step.handle_message(msg)
+
+        # Function invoked with correct arguments
+        mock_kernel.invoke.assert_awaited_once()
+        mock_emit_event.assert_awaited()
+
+        # After invocation, input is reset
+        assert step.inputs["func"]["param"] is None
+
+
+async def test_handle_message_raises_target_function_name_mismatch():
+    """Test handle_message raises an exception if the target function is not the one that is invocable."""
+    mock_kernel = AsyncMock(spec=Kernel)
+    mock_kernel.invoke = AsyncMock(return_value=AsyncMock(value="result"))
+
+    mock_state = KernelProcessStepState(name="my_step", id="123", state=None)
+    mock_info = KernelProcessStepInfo(inner_step_type=AsyncMock(spec=type), state=mock_state, output_edges={})
+
+    # Pretend we have two functions, and only "other_func" is fully ready
+    step = LocalStep(
+        kernel=mock_kernel,
+        step_info=mock_info,
+        event_namespace="ns",
+        step_state=mock_state,  # type: ignore
+        factories={},
+        functions={"other_func": AsyncMock(spec=KernelFunction)},
+        inputs={
+            "other_func": {"param": "ready_value"},
+        },
+        initial_inputs={
+            "other_func": {"param": None},
+        },
+        initialize_task=True,
+    )
+
+    msg = LocalMessage(
+        source_id="source",
+        destination_id="dest",
+        function_name="mismatched_func",
+        values={"param": "value"},
+    )
+
+    with pytest.raises(ProcessFunctionNotFoundException) as exc:
```

Comment:
Shouldn't this be handling a `ProcessTargetFunctionNameMismatchException` per the test name?